### PR TITLE
Clean up deprecated kubernetes methods

### PR DESCRIPTION
**Pull request by ialidzhikov**
**What this PR does / why we need it**:
With #583 (staring from 0.16.0) some methods from `pkg/client/kubernetes` were deprecated in favour of controller-runtime client. This PR cleans up the deprecated methods and uses controller-runtime client as replacement.

/kind api-change
/kind cleanup

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```action developer
Deprecated methods from `pkg/client/kubernetes` are now removed. Consider to switch your invocations to [kubernetes-sigs/controller-runtime](https://github.com/kubernetes-sigs/controller-runtime) client (`.Client()`) or [kubernetes/client-go](https://github.com/kubernetes/client-go) (`.Kubernetes()`).
```

### CODE REVIEW

File: pkg/operation/garden/garden.go [hunk 1]
```diff
@@ -207,13 +211,24 @@ func VerifyInternalDomainSecret(k8sGardenClient kubernetes.Interface, numberOfSh
...
-		if _, err := k8sGardenClient.CreateConfigMap(common.GardenNamespace, common.ControllerManagerInternalConfigMapName, map[string]string{
-			common.GardenRoleInternalDomain: currentDomain,
-		}, true); err != nil {
+		configMap := &corev1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      common.ControllerManagerInternalConfigMapName,
+				Namespace: common.GardenNamespace,
+			},
+		}
+		if err := kutil.CreateOrUpdate(context.TODO(), k8sGardenClient.Client(), configMap, func() error {
```
Comment (rfranzke) [inline pkg/operation/garden/garden.go]:
```suggestion
		return kutil.CreateOrUpdate(context.TODO(), k8sGardenClient.Client(), configMap, func() error {
```

File: pkg/operation/hybridbotanist/addons.go [hunk 2]
```diff
@@ -229,16 +243,22 @@ func (b *HybridBotanist) generateOptionalAddonsChart() (*chartrenderer.RenderedC
 	// "* Addon will not be deleted when the manifest file is deleted from the $ADDON_PATH."
 	// --> As we used the 'addonmanager.kubernetes.io/mode=EnsureExists' label for the Heapster deployment in previous versions we have to delete it ourselves now.
 	//     This behavior can be removed in a future release.
-	heapsterDeployments, err := b.K8sShootClient.ListDeployments(metav1.NamespaceSystem, metav1.ListOptions{
```
Comment (rfranzke) [inline pkg/operation/hybridbotanist/addons.go]: I think this code is removed in `master`, please rebase

File: pkg/operation/hybridbotanist/addons.go [hunk 3]
```diff
@@ -229,16 +243,22 @@ func (b *HybridBotanist) generateOptionalAddonsChart() (*chartrenderer.RenderedC
...
 	if err != nil {
 		return nil, err
 	}
 	for _, deployment := range heapsterDeployments.Items {
-		if err := b.K8sShootClient.DeleteDeployment(metav1.NamespaceSystem, deployment.Name); err != nil && !apierrors.IsNotFound(err) {
+		fmt.Println("inside")
+		if err := b.K8sShootClient.Client().Delete(context.TODO(), &deployment, kubernetes.DefaultDeleteOptionFuncs...); client.IgnoreNotFound(err) != nil {
 			return nil, err
 		}
+		fmt.Println("inside 2")
```
Comment (rfranzke) [inline pkg/operation/hybridbotanist/addons.go]: leftover

File: pkg/operation/hybridbotanist/addons.go [hunk 4]
```diff
@@ -229,16 +243,22 @@ func (b *HybridBotanist) generateOptionalAddonsChart() (*chartrenderer.RenderedC
...
+		client.MatchingLabels(map[string]string{
+			"chart":  "heapster-0.1.1",
+			"origin": "gardener",
+		}))
 	if err != nil {
 		return nil, err
 	}
 	for _, deployment := range heapsterDeployments.Items {
-		if err := b.K8sShootClient.DeleteDeployment(metav1.NamespaceSystem, deployment.Name); err != nil && !apierrors.IsNotFound(err) {
+		fmt.Println("inside")
```
Comment (rfranzke) [inline pkg/operation/hybridbotanist/addons.go]: leftover

File: pkg/operation/hybridbotanist/cloud_config.go [hunk 5]
```diff
@@ -319,7 +326,16 @@ func (b *HybridBotanist) computeBootstrapToken() (secret *corev1.Secret, err err
 			bootstraptokenapi.BootstrapTokenUsageAuthentication: []byte("true"),
 			bootstraptokenapi.BootstrapTokenUsageSigningKey:     []byte("true"),
 		}
-		return b.K8sShootClient.CreateSecret(metav1.NamespaceSystem, secretName, bootstraptokenapi.SecretTypeBootstrapToken, data, true)
+
+		if err := kutil.CreateOrUpdate(context.TODO(), b.K8sShootClient.Client(), secret, func() error {
```
Comment (rfranzke) [inline pkg/operation/hybridbotanist/cloud_config.go]:
```suggestion
		err := kutil.CreateOrUpdate(context.TODO(), b.K8sShootClient.Client(), secret, func() error {
			...
		}
		return secret, err
```

File: pkg/operation/hybridbotanist/controlplane.go [hunk 6]
```diff
@@ -205,7 +227,13 @@ func (b *HybridBotanist) DeployETCD() error {
 				return err
 			}
 		}
-		if err := b.K8sSeedClient.DeleteService(b.Shoot.SeedNamespace, fmt.Sprintf("etcd-%s", role)); err != nil && !apierrors.IsNotFound(err) {
+		service := &corev1.Service{
```
Comment (rfranzke) [inline pkg/operation/hybridbotanist/controlplane.go]: Can you remove this whole `DeleteService` code? It's no longer needed, it was needed a few months ago during a migration.

File: pkg/operation/hybridbotanist/controlplane.go [hunk 7]
```diff
@@ -298,24 +326,26 @@ func (b *HybridBotanist) DeployCloudProviderConfig() error {
 // RefreshCloudProviderConfig asks the Cloud Botanist to refresh the cloud provider config in case it stores
 // the cloud provider credentials. The Cloud Botanist is expected to return the complete updated cloud config.
 func (b *HybridBotanist) RefreshCloudProviderConfig() error {
-	currentConfig, err := b.K8sSeedClient.GetConfigMap(b.Shoot.SeedNamespace, common.CloudProviderConfigName)
+	configMap := &corev1.ConfigMap{}
+	err := b.K8sSeedClient.Client().Get(context.TODO(), kutil.Key(b.Shoot.SeedNamespace, common.CloudProviderConfigName), configMap)
 	if err != nil {
```
Comment (rfranzke) [inline pkg/operation/hybridbotanist/controlplane.go]: These functions here are removed, please rebase

**Review by rfranzke — CHANGES_REQUESTED** [review]

**Comment by ialidzhikov** [comment]
Thank you for your review! Applied the suggestions.
Just found an issue with the PR, let me try to resolve it.

/hold

File: pkg/operation/botanist/secrets.go [hunk 8]
```diff
@@ -730,19 +741,34 @@ func (b *Botanist) DeployCloudProviderSecret() error {
...
 		return err
 	}
 	return nil
 }
 
 func (b *Botanist) fetchExistingSecrets() (map[string]*corev1.Secret, error) {
-	secretList, err := b.K8sSeedClient.ListSecrets(b.Shoot.SeedNamespace, metav1.ListOptions{})
-	if err != nil {
+	secretList := &corev1.SecretList{}
+	if err := b.K8sSeedClient.Client().List(context.TODO(), secretList); err != nil {
```
Comment (ialidzhikov) [inline pkg/operation/botanist/secrets.go]: This sould be
```suggestion
	if err := b.K8sSeedClient.Client().List(context.TODO(), secretList, client.InNamespace(b.Shoot.SeedNamespace)); err != nil {
```

**Review by rfranzke — APPROVED** [review]
/lgtm

**Pull request merged**